### PR TITLE
Dump translations with added spaces in redactions to a Google Sheet

### PR DESCRIPTION
**Pull request by bethanyaconnor**
This implements a very simple regex to identify spaces between parts of redaction (i.e. `[] [0]`). We hope to replace this simplistic method with something more robust and complete in the next couple of weeks. 

This PR also implements a way to dump all of these strings into a centralized location, which will be maintained for the medium term. Ideally, I would like to be able to dump these strings into a Slack channel but I suspect there will be too many strings to be able to do that right now (especially once we check for missing or extra redactions). The method in this PR identified 96 malformed redactions in es-MX alone, which would make for a very noisy Slack channel over 87 languages.

### CODE REVIEW

File: bin/i18n/i18n_script_utils.rb [hunk 1]
```diff
@@ -148,6 +170,14 @@ def redact(source, dest, *plugins)
   end
 end
 
+def contains_malformed_link_or_image(translation)
```
Comment (Hamms) [inline bin/i18n/i18n_script_utils.rb:177]: I'd love to see some comments explaining what these regexes are looking for and why we're looking for them

File: bin/i18n/sync-codeorg-out.rb [hunk 2]
```diff
@@ -36,6 +36,24 @@ def rename_from_crowdin_name_to_locale
...
+  data =
+    if is_json
+      file = File.open(file_path, 'r')
+      JSON.load(file)
+    else
+      YAML.load_file(file_path)
+    end
+
+  return unless data&.values&.first&.length
+  crowdin_file_path_match = file_path.match(/i18n\/locales\/[a-z]+\-[A-Z]+\/(.+)/)
```
Comment (Hamms) [inline bin/i18n/sync-codeorg-out.rb]: it's not clear to me why this is necessary. Can we not just use the `file_path`?

**Review by Hamms — APPROVED** [review]
This looks great! Can you share an example of what the generated spreadsheet looks like?

**Comment by bethanyaconnor** [comment]
https://docs.google.com/spreadsheets/d/1BVA-Er-xQfbBhlZkGIWIAoBn-COkBfWkX1KLgZQx6is/edit?usp=sharing. Definitely open to suggestions to what info to include in the spreadsheet!

File: bin/i18n/sync-codeorg-out.rb (hunk 2 above)
Comment (bethanyaconnor) [reply bin/i18n/sync-codeorg-out.rb]: I was trying to get rid of the `i18n/locales/` part of the filepath but now that I think about it, it's probably too complex for what it's worth. I'll remove this.

File: bin/i18n/i18n_script_utils.rb (hunk 1 above)
Comment (bethanyaconnor) [reply bin/i18n/i18n_script_utils.rb:177]: Done!

**Pull request merged**